include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -203,22 +204,25 @@ typename MatType::elem_type NSGA2::Optimize(
     population.resize(populationSize);
   }
 
-  // Set the candidates from the best front as the output.
-  std::vector<BaseMatType> front;
+  // Set the candidates from the Pareto Set as the output.
+  paretoSet.resize(population[0].n_rows, population[0].n_cols, fronts[0].size());
+  // The Pareto Set is stored, can be obtained via ParetoSet() getter.
+  for (size_t solutionIdx = 0; solutionIdx < fronts[0].size(); ++solutionIdx)
+    paretoSet.slice(solutionIdx) =
+        arma::conv_to<arma::mat>::from(population[fronts[0][solutionIdx]]);
 
-  for (size_t f: fronts[0])
-    front.push_back(population[f]);
-
-  bestFront.resize(front.size());
-  // bestFront is stored, can be obtained by the Front() getter.
-  std::transform(front.begin(), front.end(), bestFront.begin(),
-    [&](const BaseMatType& individual)
-      {
-        return arma::conv_to<arma::mat>::from(individual);
-      });
-
-  // Assign iterate to first element of the best front.
-  iterate = front[0];
+  // Calculate the objectives of the surviving population.
+  EvaluateObjectives(population, objectives, calculatedObjectives);
+  // Set the candidates from the Pareto Front as the output.
+  paretoFront.resize(calculatedObjectives[0].n_rows, calculatedObjectives[0].n_cols,
+      fronts[0].size());
+  // The Pareto Front is stored, can be obtained via ParetoFront() getter.
+  for (size_t solutionIdx = 0; solutionIdx < fronts[0].size(); ++solutionIdx)
+    paretoFront.slice(solutionIdx) =
+        arma::conv_to<arma::mat>::from(calculatedObjectives[fronts[0][solutionIdx]]);
```

Comment:
Can you use `{}` technically it's a single line, but I think with `{}` it's easier to read.